qdrant_client/local/payload_value_extractor.py

```diff
@@ -18,46 +25,46 @@ def value_by_key(payload: dict, key: str, flat: bool = True) -> Optional[List[An
     Returns:
         List of values or None if key not found.
     """
-    keys = key.split(".")
+    keys = parse_json_path(key)
     result = []
 
-    def _get_value(data: Any, k_list: List[str]) -> None:
+    def _get_value(data: Any, k_list: List[JsonPathItem]) -> None:
         if not k_list:
             return
 
-        k = k_list.pop(0)
+        current_key = k_list.pop(0)
         if len(k_list) == 0:
-            if k not in data:
-                return
-
-            value = data[k]
-            if isinstance(value, list) and flat:
-                result.extend(value)
-            else:
-                result.append(value)
-            return
+            if isinstance(data, dict) and current_key.item_type == JsonPathItemType.KEY:
+                if current_key.key in data:
+                    value = data[current_key.key]
+                    if isinstance(value, list) and flat:
+                        result.extend(value)
+                    else:
+                        result.append(value)
 
-        if not isinstance(data, dict):
-            return
+            elif isinstance(data, list):
+                if current_key.item_type == JsonPathItemType.WILDCARD_INDEX:
+                    result.extend(data)
+
+                elif current_key.item_type == JsonPathItemType.INDEX:
+                    assert current_key.index is not None
+
+                    if current_key.index < len(data):
+                        result.append(data[current_key.index])
 
-        if k.endswith("]"):
-            k, index = k.split("[")
-            data = data.get(k)
-
-            if not isinstance(data, list):
-                return
-
-            index = index.strip("]")
-            if index == "":
-                for item in data:
-                    _get_value(item, k_list.copy())
-            else:
-                i = int(index)
-                if i < len(data):
-                    _get_value(data[i], k_list.copy())
-        else:
-            if k in data:
-                _get_value(data[k], k_list.copy())
+        elif current_key.item_type == JsonPathItemType.KEY:
+            if current_key.key in data:
+                _get_value(data[current_key.key], k_list.copy())
+
+        elif current_key.item_type == JsonPathItemType.INDEX:
+            assert current_key.index is not None
+
+            if current_key.index < len(data):
+                _get_value(data[current_key.index], k_list.copy())
+
+        elif current_key.item_type == JsonPathItemType.WILDCARD_INDEX:
+            for item in data:
+                _get_value(item, k_list.copy())
```

Comment:
Need to check if `isinstance(data, dict)`/`isinstance(data, list)` here.

Reply:
I'll take a look at it

Reply:
I fixed it, good catch, thanks